### PR TITLE
3994 appointment filters

### PR DESCRIPTION
**Pull request by kpethtel**
## Description of change
This mimics the web's front end filters in our backend because we can't add them to our front end for older clients.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#3994

## Things to know about this PR
This should have no impact on production at this time because we're not using v2 on production. I will spot test this on the review instance. We also have a ticket for writing a script that will compare the appointments in web against the results of our back end.

### CODE REVIEW

File: modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb [hunk 1]
```diff
@@ -187,7 +187,7 @@
         expect(response.body).to match_json_schema('VAOS_v2_appointments')
 
         uniq_statuses = response.parsed_body['data'].map { |appt| appt.dig('attributes', 'status') }.uniq
-        expect(uniq_statuses).to match_array(%w[BOOKED CANCELLED])
```
Comment (kpethtel) [inline modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb:190]: I can only assume all the cancelled appointments were more than 30 days in the past.

File: modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb [hunk 2]
```diff
@@ -217,7 +217,7 @@
           end
         end
         expect(response).to have_http_status(:ok)
-        expect(JSON.parse(response.body)['data'].size).to eq(1305)
+        expect(JSON.parse(response.body)['data'].size).to eq(980)
```
Comment (kpethtel) [inline modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb:220]: If I had to guess, I'd speculate the missing 325 results are the result of either appointment requests with the "booked" status or perhaps appointments with missing dates. Something along those lines.

File: modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb [hunk 3]
```diff
@@ -187,7 +187,7 @@
         expect(response.body).to match_json_schema('VAOS_v2_appointments')
 
         uniq_statuses = response.parsed_body['data'].map { |appt| appt.dig('attributes', 'status') }.uniq
-        expect(uniq_statuses).to match_array(%w[BOOKED CANCELLED])
+        expect(uniq_statuses).to match_array(%w[BOOKED])
```
Comment (kpethtel) [inline modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb:190]: I can only assume all the cancelled appointments were more than 30 days in the past.

File: modules/mobile/app/services/mobile/v2/appointments/presentation_filter.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        def presentable_upcoming_appointment?(appointment)
+          appointment[:start] >= @now
+        end
+
+        def presentable_past_appointment?(appointment)
+          appointment[:start] < @now && appointment[:status] != 'cancelled'
+        end
+
+        def presentable_cancelled_appointment?(appointment)
+          appointment[:status] == 'cancelled' && appointment[:start] >= 30.days.ago.beginning_of_day
```
Comment (aherzberg) [inline modules/mobile/app/services/mobile/v2/appointments/presentation_filter.rb:35]: Some of these conditions look to be good common sense safe guards but others such as only showing cancelled appointments within the last 30 days look to be more of a web UX decision rather then a valid data safe guard. should we run it by product to make sure we want to mirror this?

**Review by aherzberg — APPROVED** [review]
Looks good! just wanted to ensure that we want to mirror all of VAOS logic. I know not doing so would make cross referencing web and the app that much harder but we haven't been afraid to diverge from VAOS logic in the past so wanted to double check.

**Pull request merged**